### PR TITLE
Added mip api for collecting info to trend database

### PR DESCRIPTION
**Pull request by henrikstranneheim**
Input: 
3 file paths from housekeeper
config_path = Path("family-id_config.yaml")
mip_config_raw = ruamel.yaml.safe_load(config_path.open())

qcmetrics_path = Path("family-id_qc_metrics.yaml")
qcmetrics_raw = ruamel.yaml.safe_load(qcmetrics_path.open())

sampleinfo_path = Path("family-id_qc_sample_info.yaml")
sampleinfo_raw = ruamel.yaml.safe_load(sampleinfo_path.open())

returns:
```
Dict: {
    'analysis_sex': {
        'ACC3243A4': 'female'
    },
    'family': 'inox',
    'duplicates': {
        'ACC3243A4': '6.523%'
    },
    'genome_build': 'GRCh37',
    'mapped_reads': {
        'ACC3243A4': '98.072%'
    },
    'mip_version': 'v5.0.11',
    'samples': ['ACC3243A4']
}```

### CODE REVIEW

File: trailblazer/mip/trending.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    """Parse the output analysis files from MIP for adding info
+    to trend database
+
+    Args:
+        mip_config_raw (dict): raw YAML input from MIP analysis config file
+        qcmetrics_raw (dict): raw YAML input from MIP analysis qc metric file
+        sampleinfo_raw (dict): raw YAML input from MIP analysis qc sample info file
+    Returns:
+        dict: parsed data
+    """
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: yay for documentation!

File: trailblazer/mip/trending.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    ## Add duplicates from qc_metric file
+    for sample_data in qcmetrics_data['samples']:
+         duplicates_percent = sample_data['duplicates'] * 100
+         outdata['duplicates'][sample_data['id']] = f"{duplicates_percent:.3f}%"
+
+    ## Add mapped read from qc_metric file
+    for sample_data in qcmetrics_data['samples']:
+         mapped_reads_percent = sample_data['mapped'] * 100
+         outdata['mapped_reads'][sample_data['id']] = f"{mapped_reads_percent:.3f}%"
```
Comment (ingkebil) [inline trailblazer/mip/trending.py]: yay for python3 style formatting!

**Comment by ingkebil** [comment]
It does make sense to put yaml and not file names for testing purposes :)

**Comment by henrikstranneheim** [comment]
True. I merely followed the convention of what Robin imported into the functions in files.py. I think this would do quite nicely as long as it is easy to implement via cg. Also cg need to use the family adn sample ids of the customer and not our internal ones. I tag these along for mapping purpose later.

**Comment by ingkebil** [comment]
To make sure the interface is crystal clear, it would be nice to emphasize what you need through the variable names. We usually use sample_id (or id in sample context) for the internal sample id and sample_name (or name in sample context) for the external or customer name. (Based on how LIMS names those properties.)

This is btw the "backend" of cgadmin, the function that aggregated the data and fed it into cgadmin's delivery report: https://github.com/Clinical-Genomics/cgstats/blob/master/cgstats/analysis/export.py

And the models of the data captured: https://github.com/Clinical-Genomics/cgstats/blob/master/cgstats/analysis/models.py

**Comment by ingkebil** [comment]
The automatic generation of the delivery report worked like this:
HK was asked about the recent cases, those were fed into cglims to fetch lab specific information + pedigree. That result was fed into `cgstats anaysis export`, which generated json for cgadmin. cgadmin created the delivery report.

**Comment by henrikstranneheim** [comment]
Alright. I will merge this now and let's iterate when trying to use the API

**Pull request merged**